### PR TITLE
Removing the last packages.config and replacing them with project.json

### PR DESCRIPTION
**Pull request by tannergooding**
FYI. @jasonmalinowski, @dotnet/roslyn-infrastructure

### CODE REVIEW

**Comment by tannergooding** [comment]
This removes the last of the packages.config files, replacing them with project.json. This also removes the packages.config restore directory, which was `../packages` and resulted in files being created outside the repo that would not be cleaned by `git clean -xdf`

File: NuGet.Config [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
          is newer. To prevent that, turn off package restore. -->
-    
     <add key="automatic" value="false" />
   </packageRestore>
   <solution>
     <add key="disableSourceControlIntegration" value="true" />
   </solution>
   <config>
-    <add key="repositoryPath" value="../packages" />
+    <add key="dependencyVersion" value="HighestMinor" />
```
Comment (jasonmalinowski) [inline NuGet.Config]: Why this change? That seems unexpected.

File: src/Tools/Github/GitMergeBot/packages.config [hunk 2]
```diff
@@ -1,9 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<packages>
-  <package id="LibGit2Sharp" version="0.22.0" targetFramework="net46" />
```
Comment (jasonmalinowski) [inline src/Tools/Github/GitMergeBot/packages.config]: You need to update the project file too.

Comment (tannergooding) [inline src/Tools/Github/GitMergeBot/packages.config]: Realized what you meant, updated.

**Comment by tannergooding** [comment]
Updated to remove any references from the proj files that were added by the packages.config

**Comment by tannergooding** [comment]
windows_eta_open_prtest is failing due to the required closed side changes linked above.

**Comment by tannergooding** [comment]
Updating the SyntaxVisualizer to use the v2.0 Roslyn NuGet packages after investigation and discussion with @jmarolf

**Comment by tannergooding** [comment]
@jmarolf, we can still target v1.0 if we explicitly reference v1.1.0 of the Analyzers (which we were already including in the SDK). I have opted to do that instead as it makes the process simpler.

If we still want to move everything to v2.0, we can do that, but should do so in a separate PR.
